src/common/util.cc

```diff
@@ -109,6 +110,15 @@ static bool os_release_parse(map<string, string> *m, CephContext *cct)
   file_values_parse(kvm, fp, m, cct);
 
   fclose(fp);
+#endif
+#if defined(FREEBSD)
+  struct utsname u;
+  r = uname(&u);
+  if (r >= 0) {
```

Comment:
should be
```c
if (r == 0) {
```
or 
```c
if (!r) {
```

Reply:
and where is the closing curly bracket ?

Reply:
@tchaikov 
Had more errors `FREEBSD`  needs to be `__FREEBSD__` .
So my compilation did not catch this.